### PR TITLE
FailedChannelState on error message

### LINKED ISSUE #238: RTL14: ERROR message for channel
Write spec tests for [RTL14](https://docs.ably.io/client-lib-development-guide/features/#RTL14), ensure they pass and that the affected API conforms to [the 1.2 API design](https://github.com/ably/ably-go/issues/197).

### PR DESCRIPTION
**Pull request by sacOO7**
- Added test to check if channel transitioned to failed state on msg with error action
- Fixed #238

### CODE REVIEW

File: ably/realtime_channel_spec_test.go [hunk 1]
```diff
@@ -711,3 +711,87 @@ func TestRealtimeChannel_RTL13_HandleDetached(t *testing.T) {
...
+
+		err := ablytest.Wait(ablytest.ConnWaiter(c, c.Connect, ably.ConnectionEventConnected), nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		channel = c.Channels.Get("test")
+
+		ctx, cancel := context.WithCancel(context.Background())
+		go channel.Attach(ctx)
```
Comment (tcard) [inline ably/realtime_channel_spec_test.go:755]: As explained [here](https://github.com/ably/ably-go/pull/288#discussion_r587259552), please don't use plain `go` in tests.

File: ably/realtime_channel_spec_test.go [hunk 2]
```diff
@@ -711,3 +711,87 @@ func TestRealtimeChannel_RTL13_HandleDetached(t *testing.T) {
...
+
+		var change ably.ChannelStateChange
+		ablytest.Instantly.Recv(t, &change, stateChanges, t.Fatalf)
+		if expected, got := ably.ChannelStateFailed, change.Current; expected != got {
+			t.Fatalf("expected %v; got %v (event: %+v)", expected, got, change)
+		}
+		if got := fmt.Sprint(change.Reason); !strings.Contains(got, errInfo.Message) {
+			t.Fatalf("expected %+v; got %v (error: %+v)", errInfo, got, change.Reason)
+		}
+
```
Comment (tcard) [inline ably/realtime_channel_spec_test.go:792]: You're missing this part:

```go
"and the Channel.errorReason should be set"
```

A check on `Channel.state` would be good too for completion.

**Review by tcard — CHANGES_REQUESTED** [review]

File: ably/realtime_channel_spec_test.go (hunk 1 above)
Comment (sacOO7) [reply ably/realtime_channel_spec_test.go:755]: Actually, it's being used in another setup script too
 https://github.com/ably/ably-go/blob/5b39ec35624c26f76c9bf036d4fe8b1b2d334ce8/ably/realtime_channel_spec_test.go#L512

Comment (sacOO7) [reply ably/realtime_channel_spec_test.go:755]: Should it be changed there too?

File: ably/realtime_channel_spec_test.go (hunk 2 above)
Comment (sacOO7) [reply ably/realtime_channel_spec_test.go:792]: Ok, I will check that one

File: ably/realtime_channel_spec_test.go (hunk 1 above)
Comment (tcard) [reply ably/realtime_channel_spec_test.go]: @sacOO7 Yes, certainly. It doesn't need to be part of this PR though.

**Review by tcard — APPROVED** [review]

**Review by lmars — APPROVED** [review]

**Pull request merged**